packages/common/src/items/create-content-with-file.ts

```diff
@@ -0,0 +1,96 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import {
+  addItemPart,
+  cancelItemUpload,
+  commitItemUpload,
+  createItem,
+  ICreateItemResponse,
+} from "@esri/arcgis-rest-portal";
+import { IItemAdd } from "@esri/arcgis-rest-types";
+import { cloneObject } from "../util";
+import { batch } from "../utils";
+import { _prepareUploadRequests } from "./_internal/_prepare-upload-requests";
+
+/**
+ * Creates an item in online from a local file/item.
+ * Upload is multithreaded as the item is chunked up.
+ *
+ * @export
+ * @param {IItemAdd} item Item to be uploaded into online.
+ * @param {IUserRequestOptions} requestOptions
+ * @return {*}  {Promise<string>} Newly created item id
+ */
+export async function createContentWithFile(
```

Comment:
I'd not considered this before, but the name `createContentWithFile` seems to imply it'll create/resolve an object that implements IHubContent interface, though it resolves `ICreateItemResponse`. Maybe this method would be better named `createItemWithFile` since it lives in the `items` namespace, and it's returning the result of the RESTJS `createItem` function?